product-scenarios/code-coverage/code-coverage.sh

```diff
@@ -0,0 +1,164 @@
+#!/bin/bash
+
+# Copyright (c) 2019, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#=== FUNCTION ==================================================================
+# NAME: generate_code_coverage
+# DESCRIPTION: Genaret code coverage reports.
+# PARAMETER 1: Path to input directory
+# PARAMETER 2: Path to output directory
+#===============================================================================
+
+#set -o xtrace
+
+VAR_TEST_PLAN_ID=""
+VAR_IS_TESTGRID=""
+VAR_TINKERER_ENDPOINT=""
+VAR_TINKERER_USERNAME=""
+VAR_TINKERER_PASSWORD=""
+VAR_REM_DIR=""
+
+function generate_code_coverage(){
+    INPUT_DIR=$1
+    OUTPUT_DIR=$2
+    HOME=`pwd`
+
+    #=============== Read Deployment.property file ===============================================
+
+    VAR_TEST_PLAN_ID=$(get_prop 'TEST_PLAN_ID')
+    VAR_IS_TESTGRID=$(get_prop 'IS_TESTGRID')
+    VAR_TINKERER_ENDPOINT=$(echo $(get_prop 'TESTGRID_TINKERER_ENDPOINT') | sed 's/\\//g')
+    VAR_TINKERER_USERNAME=$(get_prop 'TESTGRID_TINKERER_USERNAME')
+    VAR_TINKERER_PASSWORD=$(get_prop 'TESTGRID_TINKERER_PASSWORD')
+    VAR_REM_DIR=$(get_prop 'RemoteProductDir')
+    user=ubuntu
+
+    #=============== Code Coverage Report Generation ===========================================
+
+    #Copy files to the client side through tinkerer
+    if [ "$VAR_IS_TESTGRID" = True ] || [ "$VAR_IS_TESTGRID" = true ]; then
+
+        #Get IP address
+        echo "Get the IP address of client"
+        ipaddress=$(echo `curl http://169.254.169.254/latest/meta-data/public-ipv4`)
```

Comment:
Normally this curl command is used to get the IP address of the any AWS instances. It doesn't change time to time. Because it's not depending on the instance I have used it here.